apps/web/src/pages/FriendsPage/FriendsPage.tsx

```diff
@@ -42,120 +44,112 @@ export default function FriendsPage(): JSX.Element {
-          data.data.errors[0].status === 500
-        ) {
-          setDisplayError('Email does not exist');
-        }
-      }
-    } else {
-      setDisplayError('Please try again');
+    if (!EMAIL_REGEX.test(formValues.email)) {
+      return setDisplayError('Please try again');
     }
 
-    const data = await api.post(
+    const {
+      data: { errors },
+    } = await api.post(
       '/users/send-friend-requests/create',
       JSON.stringify({ sendToUsersByEmail: [formValues.email] })
     );
-    console.log(data);
+
+    if (errors.length > 0) {
+      console.log(errors);
+      // TODO: Fix the error that is returned from the api call
+      if (errors[0].status === 500) {
+        setDisplayError('Email does not exist');
+      }
+    }
   };
 
   const onAcceptFriendRequestHandler = async (fromUid: UserId) => {
-    const data = await api.post(
+    const { data } = await api.post(
       '/users/receive-friend-requests/update-status',
       JSON.stringify({ status: FRIEND_REQUEST_STATUS.ACCEPTED, fromUid })
     );
     console.log(data);
   };
 
   const onDeclineFriendRequestHandler = async (fromUid: UserId) => {
-    const data = await api.post(
+    const { data } = await api.post(
       '/users/receive-friend-requests/update-status',
       JSON.stringify({ status: FRIEND_REQUEST_STATUS.DECLINED, fromUid })
     );
     console.log(data);
   };
 
   const onDeleteFriendRequestHandler = async (friendUid: UserId) => {
-    const data = await api.post(
+    const { data } = await api.post(
       '/users/friends/delete',
       JSON.stringify({ friendByUserId: friendUid })
     );
     console.log(data);
   };
 
-  React.useEffect(() => {
-    if (userRecord) {
-      const { uid } = userRecord;
-
-      // Get friends
-      const unsubscribeFriendSnapshot = getAllSubCollDocsSnapshot$(
-        `/users/${uid}/friends/`,
-        {
-          next: (friendsSnapshot) => {
-            setFriends(friendsSnapshot.docs.map((doc) => doc.data() as Friend));
-          },
-        }
-      );
-
-      // Get received friend requests
-      const unsubscribeReceiveFriendRequestSnapshot =
-        getAllSubCollDocsSnapshot$(`/users/${uid}/receive-friend-requests/`, {
-          next: (receiveFriendRequestsSnapshot) => {
-            setReceivedFriendRequests(
-              receiveFriendRequestsSnapshot.docs.map(
-                (doc) => doc.data() as ReceiveFriend
-              )
-            );
-          },
-        });
-
-      // Get sent friend requests
-      const unsubscribeSendFriendRequestSnapshot = getAllSubCollDocsSnapshot$(
-        `/users/${uid}/send-friend-requests/`,
-        {
-          next: (sendFriendRequestsSnapshot) => {
-            setSentFriendRequests(
-              sendFriendRequestsSnapshot.docs.map(
-                (doc) => doc.data() as SendFriend
-              )
-            );
-          },
-        }
-      );
-
-      // Not sure about this
-      return () => {
```

Comment:
This wouldn't properly unsubscribe from the observable and could cause a memory leak. Better to split into 3 separate hooks.

Reply:
@ivancuadra869 Thinking about this again and this would properly subscribe from all the observables!